clang/lib/CIR/CodeGen/CIRRecordLayoutBuilder.cpp

```diff
@@ -628,6 +629,24 @@ void CIRRecordLowering::accumulateFields() {
   }
 }
 
+void CIRRecordLowering::calculateZeroInit() {
+  for (const MemberInfo &member : members) {
+    if (member.kind == MemberInfo::InfoKind::Field) {
+      if (!member.fieldDecl || isZeroInitializable(member.fieldDecl))
+        continue;
+      IsZeroInitializable = IsZeroInitializableAsBase = false;
```

Comment:
I think `IsZeroInitializable` and `IsZeroInitializableAsBase` are a bit odd. They're only really used within `CIRGenTypes::computeRecordLayout`. I wonder if we can remove these two member variables and replace them with a function returning a `bool` so there's no/less partially initialized state in this class?